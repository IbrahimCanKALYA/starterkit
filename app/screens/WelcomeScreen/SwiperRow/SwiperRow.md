# SwiperRow

## Description

- Returns a component which will be displayed on a single swiper row.

## Funtions

- absent !

## Props

- text(String) : Text will be displayed center of a swiper component.

- button(Boolean) : Its decides component will have a button or not.

- buttonText(String) : Text for displaying center of button.

- onPressButton(Function) : Function will be triggered when button pressed.