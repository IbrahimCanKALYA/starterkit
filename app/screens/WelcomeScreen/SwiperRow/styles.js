/* @flow */
const styles = {
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonStyle: {
    marginHorizontal: 20,
    position: 'absolute',
    bottom: 10,
  },
  buttonTextStyle: {
    fontWeight: 'bold',
    color: '#FFF',
  },
};

export default styles;
