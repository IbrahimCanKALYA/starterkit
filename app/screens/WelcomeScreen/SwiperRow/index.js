/* @flow */
import React, { Component } from 'react';
import { View, Text } from 'react-native';
import propTypes from 'prop-types';

import { Button } from 'components';
import styles from './styles';

type Props = any;
class SwiperRow extends Component<Props> {
  static defaultProps: any;

  render() {
    const {
      props: {
        text, button, buttonText, onPressButton,
      },
    } = this;
    return (
      <View style={styles.container}>
        <Text>{text}</Text>
        {button ? (
          <Button
            buttonStyle={styles.buttonStyle}
            buttonTextStyle={styles.buttonTextStyle}
            text={buttonText}
            onPress={onPressButton}
          />
        ) : null}
      </View>
    );
  }
}

SwiperRow.propTypes = {
  text: propTypes.string,
  button: propTypes.bool,
  buttonText: propTypes.string,
  onPressButton: propTypes.func,
};

SwiperRow.defaultProps = {
  text: '',
  button: false,
  buttonText: 'Start',
  onPressButton: () => {},
};

export default SwiperRow;
