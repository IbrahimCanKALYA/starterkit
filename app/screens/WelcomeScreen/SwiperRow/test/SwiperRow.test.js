import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SwiperRow from '../index';

test('SwiperRow renders correctly', () => {
  const tree = renderer
    .create(<SwiperRow text="" button={false} buttonText="Start" onPressButton={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
