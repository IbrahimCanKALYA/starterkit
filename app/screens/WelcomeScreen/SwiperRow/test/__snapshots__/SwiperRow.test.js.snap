// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwiperRow renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    
  </Text>
</View>
`;
